packages/web-components/src/components/va-date/va-date.tsx

```diff
@@ -0,0 +1,242 @@
+import {
+  Component,
+  Event,
+  EventEmitter,
+  Host,
+  Prop,
+  h,
+} from '@stencil/core';
+
+import {
+  months,
+  days,
+} from '../../../../react-components/src/helpers/options-for-select.js';
+
+@Component({
+  tag: 'va-date',
+  styleUrl: 'va-date.css',
+  shadow: true,
+})
+export class VaDate {
+  /**
+   * Render marker indicating field is required.
+   */
+  @Prop() required: boolean;
+
+  /**
+   * Label for the field.
+   */
+  @Prop() label!: string;
+
+  /**
+   * Used to create unique name attributes for each input.
+   */
+  @Prop() name!: string;
+
+  /**
+   * The error message to render (if any)
+   * This prop should be leveraged to display any custom validations needed for this component
+   */
+  @Prop() error: string;
+
+  /**
+   * Set the `min` value on the year input.
+   */
+  @Prop() minYear: number;
+
+  /**
+   * Set the `max` value on the year input.
+   */
+  @Prop() maxYear: number;
+
+  /**
+   * Set the default month on the month input.
+   */
+  @Prop() month: string;
+
+  /**
+   * Set the default day on the day input.
+   */
+  @Prop() day: string;
+
+  /**
+   * Set the default year on the year input.
+   */
+  @Prop() year: string;
+
+  /**
+   * Fires when the month input loses focus after its value was changed
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  monthChangeEvent: EventEmitter;
+
+  /**
+   * Fires when the month input loses focus
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  monthBlurEvent: EventEmitter;
+
+  /**
+   * Fires when the day input loses focus after its value was changed
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  dayChangeEvent: EventEmitter;
+
+  /**
+   * Fires when the day input loses focus
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  dayBlurEvent: EventEmitter;
+
+  /**
+   * Fires when the year input loses focus after its value was changed
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  yearChangeEvent: EventEmitter;
+
+  /**
+   * Fires when the year input loses focus
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  yearBlurEvent: EventEmitter;
```

Comment:
Are these custom events necessary? Also can we get a `dateChange` event or something? The current `Date` react component has the single event handler:

https://github.com/department-of-veterans-affairs/component-library/blob/8f70616a228f1a1a6d2f0c6980c0c1ac402e1c35/packages/react-components/src/components/Date/Date.jsx#L54